docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -384,73 +521,113 @@ status = ml_train_model_set_optimizer(model, optimizer);
 There are two ways to create a dataset. One is from a file, and the other one is from a callback.
 In either case, you need to provide streams of tensor data and arrays of values representing the label, usually one-hot-encoded.
 
-#### Set a dataset from a file
+#### Sample, batch and epoch
+
+This section explains the following three concepts: sample, batch and epoch.
+
+Let's assume a given model requires three inputs and two labels.
+
+1. Sample
+
+A sample denotes a single pair of inputs/labels.
+
+```
+[input_1][input_2][input_3][label_1][label_2]
+```
+
+2. Batch
+
+A batch is a bundle of samples which will be fed to a single iteration. If a batch is of size 5, the batch layout, where `/*` denotes nth sample will be:
+
+```
+[[input_1/1][input_1/2][input_1/3][input_1/4][input_1/5]],
+[[input_2/1][input_2/2][input_2/3][input_2/4][input_2/5]],
+[[input_3/1][input_3/2][input_3/3][input_3/4][input_3/5]],
+[[label_1/1][label_1/2][label_1/3][label_1/4][label_1/5]],
+[[label_2/1][label_2/2][label_2/3][label_2/4][label_2/5]],
+```
+
+3. Epoch
 
-To create a `dataset` from a file, a training set and a label set must be provided. A validation set and a test set can be provided if needed.
+An epoch refers to a full, exhaustive visit to a dataset.
 
-A label set is a file that has actual labels each in a line. For example:
+So, if you consider [Cifar-100](https://www.cs.toronto.edu/~kriz/cifar.html#The%20CIFAR-100%20dataset) dataset. When using 100% of training set, a sigle training epoch will be visiting 50,000 samples. If batch size is 100, it will consists of 500 batches(or iterations).
```

Comment:
Could you verify below alternative?

`When using the full training set, a single training epoch will contain 50,000 samples. If batch size is 100, it will be of 500 batches(or iterations).`

I think we can't use the word `can` it's not about `it cant be`, but it is sort of the description of the fact.